### DISCUSSION TITLE
Becoming an official app despite possible security risks

### DISCUSSION THREAD

**Discussion by stratege1401** [JonathanTreffler/sharerenamer#61] [Ideas]
The off topic guessable security topic has become a request as suggested by DecaTec !!! Justed need to wait for NC18 fix first !!! Off course.

You want security over "guesslinks", add f2a, or yubikey or password... no random odd policy from the past like the laughable "guesslink" reason...

humour mode on: in case of extra security needed against "guesslink", use a scissor... cut down the link.

**Comment by DBJRdev** [comment]
ShareRenamer is a free add-on to Nextcloud. Some people have spent a lot of effort and time to provide such tools to us users. We are grateful for their efforts! It is a pity that someone's issue report uses derogatory terms like "stupidity". I believe that such wording should be non-standard here.
Beyond the apparent differences in vocabulary: we're rying to understand the issue description, but have difficulties to fully understand what the text is about. Could you, @stratege1401, please explain it again in other words? That might help. Thank you!

**Comment by evanrich** [comment]
> ShareRenamer is a free add-on to Nextcloud. Some people have spent a lot of effort and time to provide such tools to us users. We are grateful for their efforts! It is a pity that someone's issue report uses derogatory terms like "stupidity". I believe that such wording should be non-standard here.
> Beyond the apparent differences in vocabulary: we're rying to understand the issue description, but have difficulties to fully understand what the text is about. Could you, @stratege1401, please explain it again in other words? That might help. Thank you!

I think what he's trying to say, if i can decode his rantings and gibberish, is that the "random string of characters" that currently get generated by nextcloud (without this addon) are stupid, and that if nextcloud wanted to make things "secure" they'd use 2FA, etc.  If the link ever was compromised, you could "cut" (I'm assuming deactiveate) the link to prevent anyone from further using it...which from a security standpoint is dumb, since the link was already compromised, but hey...I'm just guessing at what he said....doesn't really seem targeted at this addon exactly, more like nextcloud itself, and he's taking it out on this plugin issue page because maybe he thinks this is run by nextcloud devs...

just my 0.02.

**Comment by FelixBuehler** [comment]
The issue is about the point, that Nextcloud won't include a ShareRename-Feature by default. They argue, that this is a security "risk".
This is not wrong, but that implies, that a randomized link (/s/h3hf82if29) is more secure than a generated one (/s/myShare). Security by obscurity never worked.
If you have sensible data, you should NEVER share them without a password, no matter if the link is /s/superSecureLink oder /s/nf87sd6f78sdgf8. 
See als here:
https://github.com/nextcloud/server/issues/4536#issuecomment-412079064

**Comment by DecaTec** [comment]
OK, I've changed the title of the issue to a more meaningful wording.

First of all, let's see if and when this app can be made available for NC 18+ When this is done, we can discuss if the app really should become "official".

@jospoortvliet: Any thoughts?

**Comment by stratege1401** [comment]
apologies to all, i was not able to anwser to any of you about my "ranting" ( nice of you ) due to the pandemic situation ... thanks to Decatec, the obvious is done...

To all, stay safe !

**Comment by stratege1401** [comment]
Any news on this project ??

**Comment by intervisionlord** [comment]
NextCloud Devs politics seems so strange for me.
Isn't it right to let users decide for which data to leave generated links, and to which ones created manually?
It is possible not to give such functionality by default, but if the user is aware of the risk and needs this functionality - let it be! Users are primarily responsible for their servers and data.

**Comment by thetuxinator** [comment]
If Security responsible People at Nextcloud feel, guessable links are a security concern in a Product in which you send links to People with unencrypted Email, they really should quit their job :D :D ;-) . I know quite much about ICT-Security and have worked at a Certificate Authority Company in the past, i think if i would have told my boss about such a security measure/feature, he would have fired me :D . If Nextcloud Security responsibles have so much fear about that feature, why you allow sending and sharing links at all? Mail seems much less secure ;-) . If Nextcloud would add that feature, you may just force password-usage if it's a guessable Link, btw what does guessable mean in your term, if you go to the math  and probability side, its even less possible to guess "random lnks like" https://example.com/092340j23cjs0dsi3 if you add completely other (as Nextcloud call guessable) links like https://example.com/mysharedponydirectory as if somebody tries all completely random ones and hits somehow https://example.com/mysharedponydirectory he has to restart as he thinks his algorithm didn't work and he only found 1 link ;-) so whats more secure? https://example.com/mysharedponydirectory adds some more randomness :D :D so from my point of view +1 for this feature. Also from the "computers point of view" https://example.com/092340j23cjs0dsi3 links are more guessable than non-machine made ones, as a computer has no idea, no involved random number generator and nothing which could be compromised or broken, as we invented the link in our brain.

**Comment by Carl1991** [comment]
The whole security argument is dumb, security by obscurity isn't viable.
If I want to share a folder with pictures with friends/family it would be much simpler to have a custom share link, as for the "security" aspect, there was always an option to password protect the share

**Comment by mMuck** [comment]
I just found this and I'm unsure why this issue is still in this issue list. The NC team has their POV, everyone has one POV. If they let the new app pass to the store - there is no problem IMO. Everyone who wants the custom share link extension can have it, the others simply not install the app. The discussion can be made with the official NC team but I think won't work well. It's about the context how NC is used. Some links require link shortening (perfect with the app), some not.

~~Someone can mention at the apps' Readme.md that short and/or guessable links can possibly lead to a security problem if used for sensitive data, mentioning the password suggestion for really sensitive things.~~ It's mentioned. Oops.
So there is no (open) issue here(?). Am I wrong? 😉

**Reply by melyux** [reply]
Yeah, there is an issue, because NC apps can't change the behavior of the NC mobile app.

**Comment by thetuxinator** [comment]
> I just found this and I'm unsure why this issue is still in this issue list. The NC team has their POV, everyone has one POV. If they let the new app pass to the store - there is no problem IMO. Everyone who wants the custom share link extension can have it, the others simply not install the app. The discussion can be made with the official NC team but I think won't work well. It's about the context how NC is used. Some links require link shortening (perfect with the app), some not.
> 
> ~Someone can mention at the apps' Readme.md that short and/or guessable links can possibly lead to a security problem if used for sensitive data, mentioning the password suggestion for really sensitive things.~ It's mentioned. Oops. So there is no (open) issue here(?). Am I wrong? wink
Thats simply completely wrong! that has nothing to do with security! I worked in highest security environments for years  and if you argue like that, every link on the internet would have to be randomized! Getting headache form this issue here! If you need security on your website or NC or anywhere 1. do request user login 2. do encryption 3. do 2factor 4. do 3factor 5. do 2 or more person auth! 

let me show with an example that its complete nonsense:

Normal Internet World:
website.com -> public all ok everybody may access
website.com/downloadmysoftware -> non public requires account/login

now what nextcloud would do for the above to improve security:
website.com -> public all ok  everybody may access
website.com/downloadmysoftware -> replace the former with  website.com/0982h3f0923hf0923hf2039fhwwefh21 to make it unaccesible for others 

if you see the above, i have to ask, are you kidding me? Still somebody may share website.com/0982h3f0923hf0923hf2039fhwwefh2 and everybody getting that link may just download without and account!

my example show perfectly, NC tries to reach security by obscurity, now please some dev, prove the contrary here! bring us an example of how "your way" improves security!

Btw in the past Zoom had a perfect example why above NC way does not work, as by default meeting links where randomized but not password protected! thats not the only example!

**Comment by mMuck** [comment]
> some dev, prove the contrary here

"here" - this is my main objection. Within this issue tracker conversation there is nothing to prove.  I asked "where is the issue" (within the NC sharerenamer app) in order to close this open issue for this app project. The app has a given and simple explained functionality. There is a need for this as long as this is no official functionality.

No issue to discuss here that helps to improve the app at the moment. So please close. (When someone has the chance to close 50 % of open issues within one day? 😄)
